pypsa/optimization/optimize.py

```diff
@@ -144,6 +144,18 @@ def define_objective(n, sns):
             var = m[f"{c}-{attr}"]
             objective.append((var * cost).sum())
 
+    # marginal cost on energy storage for Stores
+    if not n.stores.empty:
+        e_marginal_cost = n.stores.e_marginal_cost
+        var = m["Store-e"]
+        objective.append((var * e_marginal_cost).sum())
```

Comment:
just thinking if the marginals costs should be weighted by snapshot weightings?

Reply:
Yes, you're right. The cost should be in €/MWh/h, thus requiring a snapshot weighting.